src/main/java/seedu/exchangecoursemapper/command/CompareMappedCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.exchangecoursemapper.command;
+
+import seedu.exchangecoursemapper.storage.Storage;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.HashSet;
+
+import static seedu.exchangecoursemapper.constants.Messages.LINE_SEPARATOR;
+
+public class CompareMappedCommand extends CheckInformationCommand {
+
+    private final Storage storage;
+
+    public CompareMappedCommand(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void execute(String userInput) {
+        String[] inputs = userInput.split("pu/");
+        if (inputs.length < 3) {
+            System.out.println("Please provide two universities in the format: " +
+                    "compare pu/<University 1> pu/<University 2>");
+            return;
+        }
+
+        String university1 = inputs[1].trim();
+        String university2 = inputs[2].trim();
+
+        List<String> allModules = storage.loadAllCourses();
+
+        // Filter mappings for each university
+        List<String> uni1Modules = allModules.stream()
+                .filter(module -> module.contains(" | " + university1 + " | "))
+                .toList();
+
+        List<String> uni2Modules = allModules.stream()
+                .filter(module -> module.contains(" | " + university2 + " | "))
+                .toList();
```

Comment:
LGTM!! 👍 Maybe SLAP can be applied to this method by abstracting out the separate functions too 😸

Reply:
Good eye! I have refactored the code with SLAP.